backend/src/test/java/ddangkong/documentation/TestCookieConfig.java

```diff
@@ -0,0 +1,20 @@
+package ddangkong.documentation;
+
+import ddangkong.aop.cookie.EncryptionUtils;
+import ddangkong.aop.cookie.RoomMemberCookieEncryptor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class TestCookieConfig {
+
+    @Bean
+    public RoomMemberCookieEncryptor roomMemberCookieEncryptor() {
+        return new RoomMemberCookieEncryptor(encryptionUtils(), "test_cookie");
+    }
+
+    @Bean
+    public EncryptionUtils encryptionUtils() {
+        return new EncryptionUtils("AES", "1234567890123456");
+    }
+}
```

Comment:
이거 그냥 테스트에 있는 application.yml을 사용할 것 같은데, 따로 `Config`로 지정해주신 이유가 있나요? (정확히 어떤 에러가 있어 도입하게 되었는지 설명해주세요~)

Reply:
> 기존의 Documentation 테스트가 @WebMvcTest 로 동작하고 있었는데 쿠키관련 로직이 AOP로 빠지게 되면서
빈을 조회하는 로직이 제대로 동작하지 않았습니다.
따라서 이를 해결하기 위해 TestCookieConfig를 추가하였고 BaseDocumentationTest에서 @Import 했습니다.

해당 테스트에만 `@Import(RoomMemberCookieAspect.class)`를 추가해도 안되나요? (실험은 안해보긴 했는데, 뭔가 더 타칸이 잘 알지 않을까 싶어서 물어봤습니다~)

Reply:
만약에 어쩔 수 없이 `TestCookieConfig`를 유지해야 한다면 application-test.yml에 있는 값을 가져다 쓰는 것이 좋을 것 같아요. 나중에 yml 값을 바꾸었는데 실제 테스트 환경에 적용이 안될 수 있을 것 같아요.

```suggestion
@Configuration
public class TestCookieConfig {

    @Value("${cookie.rejoin-key}")
    private String rejoinKey;
    
    @Value("${encrypt.algorithm}")
    private String encryptionAlgorithm;
    
    @Value("${encrypt.secret-key}")
    private String encryptionSecretKey;

    @Bean
    public RoomMemberCookieEncryptor roomMemberCookieEncryptor() {
        return new RoomMemberCookieEncryptor(encryptionUtils(), rejoinKey);
    }

    @Bean
    public EncryptionUtils encryptionUtils() {
        return new EncryptionUtils(encryptionAlgorithm, encryptionSecretKey);
    }
}
```

Reply:
당시에 `@Import(RoomMemberCookieAspect.class)` 를 활용해보긴 했는데 의도한대로 동작하지 않더라고요. 테스트가 실패했습니다. 그래서 TestCookieConfig를 정의해 직접 빈을 추가해주었습니다. 하지만 오늘 해결책을 찾아서 TestCookieConfig를 사용하지 않아도 될 것 같아요! 이 부분은 수정해서 커밋할게요!